ticketmatch.rb

```diff
@@ -4,90 +4,328 @@
 #
 # You must cd to the git repo for this to work.
 #
+
 require 'rubygems'
 require 'highline/import' # see https://github.com/JEG2/highline
 require 'json'
 
-git_from = ask('Enter Git From Rev: ')
-git_to = ask('Enter Git To Rev: ') { |q| q.default = 'master' }
+# store the basic information of a git log message
+class GitEntry
+  attr_reader :hash, :description
+  attr_accessor :revert_entry, :has_revert_parent
+
+  def initialize(hash, description)
+    @hash        = hash
+    @description = description
+    # The GitEntry that is a revert of this commit
+    @revert_entry = nil
+    # we set this in the revert entry so we can avoid printing it twice
+    @has_revert_parent = false
+  end
+
+  def <=>(another_entry)
```

Comment:
I think spaceship is supposed to return ordering, not just equality? See https://stackoverflow.com/questions/3058801/overriding-rubys-spaceship-operator